syntax = "proto3";

package com.webank.ai.fate.api.serving;
option java_outer_classname = "ServingProto";

message DataMap{
    map<string, float> floatData = 1;
    map<string, string> stringData = 2;
}

message PredictRequest{
    string partyId = 1;
    string sceneId = 2;
    string role = 3;
    string modelId = 4;
    map<string, DataMap> data = 5;
}

message PredictResponse{
    string partyId = 1;
    string sceneId = 2;
    string role = 3;
    string modelId = 4;
    map<string, DataMap> data = 5;
    string dataJson = 6;
}

service PredictionService{
    rpc predict(PredictRequest) returns (PredictResponse);
}
